python/cudf/cudf/core/frame.py

```diff
@@ -3581,64 +3582,148 @@ def _reindex(
         return self._mimic_inplace(result, inplace=inplace)
 
 
-def _get_replacement_values(to_replace, replacement, col_name, column):
+def _get_replacement_values_for_columns(
+    to_replace: Any, value: Any, columns: Dict[Any, Any]
+) -> Tuple[Dict[Any, Any], Dict[Any, Any], Dict[Any, Any]]:
```

Comment:
Can the first dict returned ever have non boolean values?


```suggestion
) -> Tuple[Dict[Any, Bool], Dict[Any, Any], Dict[Any, Any]]:
```